src/api-service/__app__/onefuzzlib/azure/nsg.py

```diff
@@ -72,6 +72,15 @@ def create_nsg(name: str, location: Region) -> Union[None, Error]:
     return None
 
 
+def list_nsgs() -> List[NetworkSecurityGroup]:
+    resource_group = get_base_resource_group()
+    network_client = get_network_client()
+    nsgs = []
```

Comment:
I'll try out your suggestions

Reply:
Ah, got it. What you did is consistent with our preference for `for` loops, which is good in context. In this case, since we _only_ append elements, I think it is possibly clearer to use my suggestion. It is also idiomatic for this codebase, since we do use list comprehensions for simple cases.

IMO, the `for` loop approach is best in this codebase if we have any nontrivial logic in each loop iterator. It would definitely break existing conventions to start using `map()` and `filter()` combinators a bunch.